File: LeetCode/Last Stone Weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones_neg = [-x for x in stones]
        
        heapq.heapify(stones_neg)
        
        while len(stones_neg) > 1:
            first = heapq.heappop(stones_neg)
            second = heapq.heappop(stones_neg)
            
            if first != second:
                heapq.heappush(stones_neg, first - second)
        
        if len(stones_neg) > 0:
            return -1*stones_neg[0]
        else:
            return 0

File: LeetCode/Last Stone Weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> q;
        
        for(int i=0; i < stones.size(); i++) {
            q.push(stones[i]);
        }
        
        while(q.size() > 1) {
            int y = q.top(); q.pop();
            int x = q.top(); q.pop();
            
            if (x == y)
                continue;
            else
                q.push(y-x);
        }
        
        if (q.empty())
            return 0;
        else
            return q.top();
    }
};